wrapper/src/test/java/integration/container/aurora/postgres/AuroraPostgresReadWriteSplittingPerformanceTest.java

```diff
@@ -0,0 +1,354 @@
+import org.junit.jupiter.params.provider.MethodSource;
+import org.postgresql.PGProperty;
+import software.amazon.jdbc.PropertyDefinition;
+import software.amazon.jdbc.plugin.readwritesplitting.ReadWriteSplittingPlugin;
+import software.amazon.jdbc.util.StringUtils;
+
+public class AuroraPostgresReadWriteSplittingPerformanceTest extends AuroraPostgresBaseTest {
+
+  private static final int REPEAT_TIMES = StringUtils.isNullOrEmpty(System.getenv("REPEAT_TIMES"))
+      ? 100
+      : Integer.parseInt(System.getenv("REPEAT_TIMES"));
+
+  private static final int EXECUTE_QUERY_TIMES = StringUtils.isNullOrEmpty(System.getenv("EXECUTE_QUERY_TIMES"))
+      ? 1000
+      : Integer.parseInt(System.getenv("EXECUTE_QUERY_TIMES"));
+
+  private static final List<PerfStatSetReadOnly> setReadOnlyPerfDataList = new ArrayList<>();
+  private static final List<PerfStatExecuteQueries> executeStatementsPerfDataList = new ArrayList<>();
+
+  private static Stream<Arguments> executeStatementsParameters() {
+    return Stream.of(
+        Arguments.of(initReadWritePluginProps()),
+        Arguments.of(initNoPluginPropsWithTimeouts())
+        );
+  }
+
+  @AfterAll
+  public static void cleanUp() throws IOException {
+    doWritePerfDataToFile(
+        "./build/reports/tests/PostgresSQL_ReadWriteSplittingPerformanceResults_SetReadOnly.xlsx",
+        setReadOnlyPerfDataList);
+    doWritePerfDataToFile(
+        "./build/reports/tests/PostgresSQL_ReadWriteSplittingPerformanceResults_ReaderLoadBalancing.xlsx",
+        executeStatementsPerfDataList);
+  }
+
+  private static void doWritePerfDataToFile(
+      String fileName,
+      List<? extends PerfStatBase> dataList)
+      throws IOException {
+    if (dataList.isEmpty()) {
+      return;
+    }
+
+    try (XSSFWorkbook workbook = new XSSFWorkbook()) {
+
+      final XSSFSheet sheet = workbook.createSheet("PerformanceResults");
+
+      for (int rows = 0; rows < dataList.size(); rows++) {
+        PerfStatBase perfStat = dataList.get(rows);
+        Row row;
+
+        if (rows == 0) {
+          // Header
+          row = sheet.createRow(0);
+          perfStat.writeHeader(row);
+        }
+
+        row = sheet.createRow(rows + 1);
+        perfStat.writeData(row);
+      }
+
+      // Write to file
+      final File newExcelFile = new File(fileName);
+      newExcelFile.createNewFile();
+      try (FileOutputStream fileOut = new FileOutputStream(newExcelFile)) {
+        workbook.write(fileOut);
+      }
+    }
+  }
+
+  protected static Properties initNoPluginPropsWithTimeouts() {
+    final Properties props = new Properties();
+    props.setProperty(PGProperty.USER.getName(), AURORA_POSTGRES_USERNAME);
+    props.setProperty(PGProperty.PASSWORD.getName(), AURORA_POSTGRES_PASSWORD);
+    props.setProperty(PGProperty.TCP_KEEP_ALIVE.getName(), Boolean.FALSE.toString());
+    props.setProperty(PGProperty.CONNECT_TIMEOUT.getName(), "30");
+    props.setProperty(PGProperty.SOCKET_TIMEOUT.getName(), "30");
```

Comment:
Do these timeouts need to be 30 seconds? Tests might run faster if you lower these values.